include/fx/gltf.h

```diff
@@ -1757,46 +1782,57 @@ namespace gltf
 
                 header.jsonHeader.chunkLength = ((jsonText.length() + 3) & (~3u));
                 const uint32_t headerPadding = static_cast<uint32_t>(header.jsonHeader.chunkLength - jsonText.length());
-                header.length = detail::HeaderSize + header.jsonHeader.chunkLength + detail::ChunkHeaderSize + binHeader.chunkLength;
-
-                std::ofstream fileData(documentFilePath, std::ios::binary);
-                if (!fileData.good())
+                header.length =
+                    detail::HeaderSize +
+                    header.jsonHeader.chunkLength +
+                    detail::ChunkHeaderSize +
+                    binHeader.chunkLength +
+                    std::accumulate
+                    (document.extraChunks.begin(),
+                     document.extraChunks.end(), 0,
+                     [&](std::size_t n, detail::Chunk const & chunk)
+                     { return n + detail::ChunkHeaderSize + chunk.header.chunkLength; });
+
+                if (!os.good())
                 {
                     throw std::system_error(std::make_error_code(std::errc::io_error));
                 }
 
                 const char spaces[3] = { ' ', ' ', ' ' };
                 const char nulls[3] = { 0, 0, 0 };
 
-                fileData.write(reinterpret_cast<char *>(&header), detail::HeaderSize);
-                fileData.write(jsonText.c_str(), jsonText.length());
-                fileData.write(&spaces[0], headerPadding);
-                fileData.write(reinterpret_cast<char *>(&binHeader), detail::ChunkHeaderSize);
-                fileData.write(reinterpret_cast<char const *>(&binBuffer.data[0]), binBuffer.byteLength);
-                fileData.write(&nulls[0], binPadding);
+                os.write(reinterpret_cast<char *>(&header), detail::HeaderSize);
+                os.write(jsonText.c_str(), jsonText.length());
+                os.write(&spaces[0], headerPadding);
+                os.write(reinterpret_cast<char *>(&binHeader), detail::ChunkHeaderSize);
+                os.write(reinterpret_cast<char const *>(&binBuffer.data[0]), binBuffer.byteLength);
+                os.write(&nulls[0], binPadding);
 
                 externalBufferIndex = 1;
+
+                for(auto & chunk : document.extraChunks)
+                {
+                    os.write(reinterpret_cast<char const *>(&chunk.header), detail::ChunkHeaderSize);
+                    os.write(reinterpret_cast<char const *>(&chunk.body[0]), chunk.header.chunkLength);

```

Comment:
Per the spec, each Chunk needs to be aligned on a 4-byte boundary so that needs to be accounted for here.  I have a question out to the spec folks about what should be used as the padding bytes though.